### PR TITLE
QE: Add testing for duplicate SCC credentials

### PR DESCRIPTION
**Pull request by mtravitzky**
## What does this PR change?

This PR adds a test for correct behavior if a user tries to enter credentials for which an entry with the same user name already exists.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- Cucumber tests were added

- [x] **DONE**

## Links

Issue(s): https://github.com/SUSE/spacewalk/issues/23479
Ports(s): https://github.com/SUSE/spacewalk/pull/23484

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

**Review by NamelessOne91 — COMMENTED** [review]
Are the credentials stored even if invalid ?

I'm asking because it seems odd to have a test checking for them to be invalid first and then checking again to see if they are already present. I'd expect that invalid credentials are never "already present".
But I might be missing some context here.

**Comment by nodeg** [comment]
Yes, they are. See
image
But you have the red `X` that says `verification failed (click to refresh)`.

**Comment by mtravitzky** [comment]
This makes sense, because the validity of credentials can continuously be tested later on, while they are not immediately tested when entering them (testing runs automatically _after_ they have been added).

**Review by NamelessOne91 — APPROVED** [review]
Thanks for the clarification.
LGTM then

**Comment by nodeg** [comment]
The message that is shown in the test would be:
image

**Review by nodeg — CHANGES_REQUESTED** [review]
One thing you/we missed is to adjust the year in the first line of the file.
```ruby
# Copyright 2017-2024 SUSE LLC
```

**Comment by mtravitzky** [comment]
Copyright line has been updated to the current year.

**Review by nodeg — APPROVED** [review]
Thank you. Be aware to squash the 2 commits into one when merging this PR.

File: testsuite/features/secondary/srv_scc_user_credentials.feature [hunk 1]
```diff
@@ -30,6 +30,22 @@ Feature: SCC user credentials in the Setup Wizard
...
+    And I enter "invalidname" as "edit-user"
+    And I enter "invalidpw" as "edit-password"
+    And I click on "Save"
+    Then the credentials for "invalidname" should be invalid
+    When I ask to add new credentials
+    And I enter "invalidname" as "edit-user"
+    And I enter "invalidpw" as "edit-password"
+    And I click on "Save"
+    Then I should see a "Credentials with this username already exist" text
+    And I click on "Cancel"
```
Comment (srbarrios) [inline testsuite/features/secondary/srv_scc_user_credentials.feature:47]: When

File: testsuite/features/secondary/srv_scc_user_credentials.feature [hunk 2]
```diff
@@ -40,3 +56,7 @@ Feature: SCC user credentials in the Setup Wizard
     And I ask to delete the credentials for "SCC user"
     And I click on "Delete" in "Are you sure you want to delete these credentials?" modal
     Then I wait until I do not see "SCC user" text
+    And I wait for the trash icon to appear for "invalidname"
```
Comment (srbarrios) [inline testsuite/features/secondary/srv_scc_user_credentials.feature:59]: When

**Review by srbarrios — APPROVED** [review]

File: testsuite/features/secondary/srv_scc_user_credentials.feature (hunk 2 above)
Comment (mtravitzky) [reply testsuite/features/secondary/srv_scc_user_credentials.feature:59]: Shouldn't this also be changed in line 55 (not my code)?
